### PR TITLE
Fix keyboard events on range sliders

### LINKED ISSUE #8911: RangeSlider stuck when modified by pressing the keyboard arrows
When changing the lower value of a RangeSlider by pressing the keyboard arrows the RangeSlider gets stuck on NaNs.

image

Bokeh version: 1.1.0
Platform: Windows
Command line: bokeh serve app

### PR DESCRIPTION
**Pull request by xavArtley**
- [x] issues: fixes #8911
- [x] tests added / passed

### CODE REVIEW

**Comment by bryevdv** [comment]
@xavArtley Thanks for looking in to this. If I can impose on you slightly, I'd appreciate taking the opportunity to improve things a bit:

* the render methods is 100+ lines long. It should really be split up at least according to some helper methods for n=1 vs n=1 but possibly other helpers can be extracted as well. 

* another option is to move the n=1 vs n=2 keypress handling code into appropriate subclasses under the same name `_handle_keypress` or whatever, then `AbstractSlider` just calls this and it delegates appropriately based on the concrete type. This would be my preference but it is a little more work. 

* The code in the switch statements is basically repeated three times, it could also be pulled into a little helper.

**Comment by xavArtley** [comment]
I added some integration tests, it works locally for me but it seems to fail on the CI build.

**Comment by bryevdv** [comment]
I can take a look soon if you don't figure it out first :)

**Comment by xavArtley** [comment]
I think I know what's going on. When both handle of the range_slider are supperposed the selenium webdriver select the wrong element

**Comment by xavArtley** [comment]
I change tests to avoid handle overlapping however it continues to fail

**Comment by bryevdv** [comment]
I'll try to take a look tonight cc @mattpap in case you have any ideas at a glance

**Comment by xavArtley** [comment]
I finally found the right method

**Comment by bryevdv** [comment]
Thank you for the hard work on this @xavArtley I will get it merged tonight

**Comment by bryevdv** [comment]
@xavArtley When I try this on chrome with the `export_csv` example app, the keyboard arrows move the slider, and show updated values in the UI, but python callbacks in the server are not getting called until I actually scrub the slider with a mouse. Do you see similar behavior?

File: bokehjs/src/lib/models/widgets/abstract_slider.ts [hunk 1]
```diff
@@ -114,6 +114,39 @@ export abstract class AbstractSliderView extends ControlView {
...
+      }
+      case 39: {
+        value[idx] = Math.min(value[idx] + step, high)
+        break
+      }
+      default:
+        return
+    }
+    if (is_range)
+      this.model.properties.value.change.emit()
```
Comment (bryevdv) [inline bokehjs/src/lib/models/widgets/abstract_slider.ts]: In order to trigger a sync with Python there will need to be an actual, real assignment to `this.model.value`

**Comment by xavArtley** [comment]
I do not really have any perspective on the conflicts that will be introduced with version 2.0. In any case it will be necessary to solve them. Here is a bug fix and it does not break the api. I think it may be interesting for people who are going to stay on version 1. of bokeh even after the release of version 2. to have the fix

**Comment by bryevdv** [comment]
OK let's go ahead and merge. I will figure the conflicts out.

**Comment by bryevdv** [comment]
@xavArtley I resolved the conflicts on the `landing-2.0` branch. But if you could look over the relevant files you changed, in the `landing-2.0` branch, just to have another pair of eyes look over things, it would be reassuring

**Pull request merged**